implementation/src/main/java/io/smallrye/metrics/setup/config/MetricsConfigurationManager.java

```diff
@@ -0,0 +1,320 @@
+            if (LOGGER.isLoggable(Level.FINEST)) {
+                LOGGER.logp(Level.FINEST, CLASS_NAME, null,
+                        "Returning \"{0}\" configuration for metric:\"{0}\" with values: {1} ",
+                        new Object[] { MP_PERCENTILES_PROP, metricName, retVal });
+            }
+            return retVal;
+        }
+        return null;
+
+    }
+
+    /**
+     * Returns the matching configuration object if it exists, null otherwise
+     * 
+     * @param metricName the metric name to check configuration against
+     * @return the matching configuration object if it exists, null otherwise
+     */
+    public synchronized HistogramBucketConfiguration getHistogrmBucketConfiguration(String metricName) {
+
+        String appName = getApplicationName();
+
+        Collection<HistogramBucketConfiguration> computedValues = histogramBucketsConfigMap.computeIfAbsent(appName,
+                f -> {
+                    Optional<String> input = ConfigProvider.getConfig().getOptionalValue(MP_HISTOGRAM_BUCKET_PROP,
+                            String.class);
+
+                    return (input.isPresent()) ? HistogramBucketConfiguration.parse(input.get()) : null;
+
+                });
+
+        if (computedValues != null && computedValues.size() != 0) {
+            HistogramBucketConfiguration retVal = HistogramBucketConfiguration.matches(computedValues, metricName);
+            if (LOGGER.isLoggable(Level.FINEST)) {
+                LOGGER.logp(Level.FINEST, CLASS_NAME, null,
+                        "Returning \"{0}\" configuration for metric:\"{1}\" with values: {1} ",
+                        new Object[] { MP_HISTOGRAM_BUCKET_PROP, metricName, retVal });
+            }
+            return retVal;
+        }
+        return null;
+
+    }
+
+    /**
+     * Returns the matching configuration object if it exists, null otherwise
+     * 
+     * @param metricName the metric name to check configuration against
+     * @return the matching configuration object if it exists, null otherwise
+     */
+    public synchronized TimerBucketConfiguration getTimerBucketConfiguration(String metricName) {
+
+        String appName = getApplicationName();
+
+        Collection<TimerBucketConfiguration> computedValues = timerBucketsConfigMap.computeIfAbsent(appName, f -> {
+            Optional<String> input = ConfigProvider.getConfig().getOptionalValue(MP_TIMER_BUCKET_PROP, String.class);
+
+            return (input.isPresent()) ? TimerBucketConfiguration.parse(input.get()) : null;
+
+        });
+
+        if (computedValues != null && computedValues.size() != 0) {
+            TimerBucketConfiguration retVal = TimerBucketConfiguration.matches(computedValues, metricName);
+            if (LOGGER.isLoggable(Level.FINEST)) {
+                LOGGER.logp(Level.FINEST, CLASS_NAME, null,
+                        "Returning \"{0}\" configuration for metric:\"{0}\" with values: {1} ",
```

Comment:
Should the parameter be `{0}, {1}, and {2}`?

Reply:
fixed